custom_components/waste_collection_schedule/waste_collection_schedule/source/kwu_de.py

```diff
@@ -0,0 +1,100 @@
+import requests
+from bs4 import BeautifulSoup
+from datetime import date
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+from waste_collection_schedule.service.ICS import ICS
+
+TITLE = "KWU Entsorgung Landkreis Oder-Spree"
+DESCRIPTION = "Source for KWU Entsorgung, Germany"
+URL = "https://www.kwu-entsorgung.de/"
+
+TEST_CASES = {
+    "Erkner": {"city": "Erkner", "street": "Heinrich-Heine-Straße", "number": "11"},
+    "Bad Saarow": {"city": "Bad Saarow", "street": "Ahornallee", "number": "1"}
+}
+
+HEADERS = {"user-agent": "Mozilla/5.0 (xxxx Windows NT 10.0; Win64; x64)"}
+
+
+class Source:
+    def __init__(self, city, street, number):
+        self._city = city
+        self._street = street
+        self._number = number
+        self._ics = ICS()
+        self._iconMap  = {
+            "Restabfall": "mdi:trash-can-outline",
+            "Gelber Sack" : "mdi:recycle",
+            "Papiertonne" : "mdi:package-variant",
+            "Biotonne": "mdi:food-apple-outline",
+        } 
+
+    def fetch(self):
+        session = requests.Session()
+
+        params = {
+            "city": self._city,
+            "street": self._street,
+            "number": self._number,
+            "direct": "true",
+        }
+                
+        r = requests.get("https://www.kwu-entsorgung.de/inc/wordpress/kal_objauswahl.php", headers=HEADERS)
+        parsed_html = BeautifulSoup(r.text, "html.parser")
+        Orte = parsed_html.find_all('option')
+
+        for Ort in Orte:
+            if self._city in Ort.text:
+                OrtValue = Ort['value']
+                break
+
+        r = requests.get("https://www.kwu-entsorgung.de/inc/wordpress/kal_str2ort.php?ort=" + OrtValue, headers=HEADERS)
```

Comment:
Please use the params argument to pass parameters in URLs: https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls
This also ensures correct encoding.

Reply:
Ok. Done. 👍